Tests/EventListener/RateLimitAnnotationListenerTest.php

```diff
@@ -452,4 +454,31 @@ public function testRateLimitThrottlingWithMessages()
         $this->assertEquals($response->getStatusCode(), 123);
         $this->assertEquals($response->getContent(), "a message");
     }
+
+    public function testRateLimitThrottlingWithFailOpen()
+    {
+        $listener = $this->createListener($this->any(), false);
+
+        $event = $this->createEvent();
+        $event->getRequest()->attributes->set('_x-rate-limit', array(
+            new RateLimit(array('limit' => 5, 'period' => 3, 'failOpen' => true)),
+        ));
+
+        $listener->onKernelController($event);
+    }
+
+    public function testRateLimitThrottlingWithFailOpenFalse()
+    {
+        $listener = $this->createListener($this->any(), false);
+        $this->expectException(\Throwable::class);
```

Comment:
Are you testing here an exception thrown by the storage or this is an exception thrown because the storage is missing?

Reply:
I think this (https://travis-ci.org/jaytaph/RateLimitBundle/jobs/600618362) ci failure is related to it

Reply:
For this one I thought testing missing storage because that will throw an error which we can test for. Do you think I use use an alternative method

Reply:
Thank you - I will have a close look